src/Base/Reader.h

```diff
@@ -57,6 +60,50 @@ class SAX2XMLReader;
 XERCES_CPP_NAMESPACE_END
 #endif
 
+
+namespace
+{
+template<typename T>
+T readerCast(const char* value);
+
+template<>
+long readerCast<long>(const char* value)
+{
+    return std::stol(value);
+}
+
+template<>
+int readerCast<int>(const char* value)
+{
+    return std::stoi(value);
+}
+
+template<>
+unsigned long readerCast<unsigned long>(const char* value)
+{
+    return std::stoul(value, nullptr);
+}
+
+template<>
+double readerCast<double>(const char* value)
+{
+    return std::stod(value, nullptr);
+}
+
+template<>
+bool readerCast<bool>(const char* value)
+{
+    return value[0] != '0';

```

Comment:
Is this sufficient? It's obviously not exhaustive (and the linter doesn't like the pointer arithmetic).

Reply:
I consider it, yes. Everything not `0` is true, likewise as in the C++ language. There are extensive tests added for the bool reading.

Reply:
It may be that we don't care, but written like this, `0xFF` will evaluate to `false` (for example).

Reply:
What about this then?
```suggestion
    return value[0] != '0' || value[1] != '\0' ;
```

Reply:
how about:
```c++
template<>
bool readerCast<bool>(const char* value)
{
    return std::string_view(value) != "0";
}
```